tasks/main.yaml

```diff
@@ -3,8 +3,7 @@
   delegate_to: "{{ groups['prometheus'][0] }}"
   blockinfile:
     dest: "{{ prometheus_config }}"
-    marker: "# {mark} ANSIBLE MANAGED BLOCK for {{ hostvars['ansible_host']|default(inventory_hostname) }}:{{ exporter_port }}"
+    marker: "# {mark} ANSIBLE MANAGED BLOCK for {{ hostvars[inventory_hostname]['ansible_host']|default(inventory_hostname) }}:{{ exporter_port }}"
     block: |
        - targets:
-           - {{ hostvars['ansible_host']|default(inventory_hostname) }}:{{ exporter_port }}
-...
+           - {{ ansible_host[inventory_hostname]|default(inventory_hostname) }}:{{ exporter_port }}
```

Comment:
this should be 
{{ hostvars[inventory_hostname]['ansible_host']|default(inventory_hostname) }}:{{ exporter_port }}

Reply:
:face_exhaling: